### PR TITLE
Change Fstype default to emptyString from ext4

### LINKED ISSUE #343: change default fstype from "ext4" to empty string
This is a followup for [issue 287](https://github.com/kubernetes-csi/external-attacher/issues/287) which adds a workaround to remove default filesystem type interference by providing a new csi-attacher argument  (--default-fstype). When this argument is not used we still default to "ext4".

Next major release allows a behavior change where we can start defaulting to empty string instead of "ext4".

Default value should change [here](https://github.com/kubernetes-csi/external-attacher/blob/dd0d1c492475dec47e6385e5178c7bd81a3d127e/cmd/csi-attacher/main.go#L69)

And warning removed [here](https://github.com/kubernetes-csi/external-attacher/blob/dd0d1c492475dec47e6385e5178c7bd81a3d127e/pkg/controller/util.go#L160)

### PR DESCRIPTION
**Pull request by Kartik494**
**What type of PR is this?**
 /kind bug

**Which issue(s) this PR fixes**:

Fixes #343 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Change Fstype default to emptyString
Action required: CSI drivers that depended on ext4 as the default need to change their deployments to explicitly set ext4.
```

### CODE REVIEW

**Comment by Kartik494** [comment]
Is this make sense to change fsType to empty string  in test case also? https://github.com/kubernetes-csi/external-attacher/blob/5545207e42af726eeb6b351bab24d7f6e11d3efb/pkg/controller/csi_handler_test.go#L47

**Comment by ggriffiths** [comment]
> Is this make sense to change fsType to empty string in test case also?
> 
> https://github.com/kubernetes-csi/external-attacher/blob/5545207e42af726eeb6b351bab24d7f6e11d3efb/pkg/controller/csi_handler_test.go#L47

Yes, let's test with the same default.

**Comment by ggriffiths** [comment]
@Kartik494 can you add a release note for this as we're changing a default value?

**Comment by jingxu97** [comment]
Just want to check whether it will have any behavior change from this?

**Comment by Kartik494** [comment]
> @Kartik494 can you add a release note for this as we're changing a default value?

Done

**Comment by Kartik494** [comment]
> > Is this make sense to change fsType to empty string in test case also?
> > https://github.com/kubernetes-csi/external-attacher/blob/5545207e42af726eeb6b351bab24d7f6e11d3efb/pkg/controller/csi_handler_test.go#L47
> 
> Yes, let's test with the same default.

Thanks for the inputs, i will modify this.

File: cmd/csi-attacher/main.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ var (
 	leaderElectionRenewDeadline = flag.Duration("leader-election-renew-deadline", 10*time.Second, "Duration, in seconds, that the acting leader will retry refreshing leadership before giving up. Defaults to 10 seconds.")
 	leaderElectionRetryPeriod   = flag.Duration("leader-election-retry-period", 5*time.Second, "Duration, in seconds, the LeaderElector clients should wait between tries of actions. Defaults to 5 seconds.")
 
-	defaultFSType = flag.String("default-fstype", "ext4", "The default filesystem type of the volume to publish.")
+	defaultFSType = flag.String("default-fstype", "", "The default filesystem type of the volume to publish.")
```
[the remark this answers is not in the document]
Comment (ggriffiths) [reply cmd/csi-attacher/main.go:69]: Yes, `Defaults to empty string` would make sense here

[the remark this answers is not in the document]
Comment (Kartik494) [reply cmd/csi-attacher/main.go]: Done

**Comment by jsafrane** [comment]
@Kartik494 please add `Action needed:` to the release note, and mention that drivers that depended on `ext4` as the default need to change their deployments to explicitly set `ext4`.

And IMO we will need to release 4.0 instead of 3.6.

**Comment by Kartik494** [comment]
> Just want to check whether it will have any behavior change from this?

Seems Like `empty string` will not change in any behaviour change but disable the interference.
Please see https://github.com/kubernetes-csi/external-attacher/pull/342#issue-1094109695

**Pull request merged**